src/hdmf/utils.py

```diff
@@ -411,7 +411,6 @@ def foo(self, **kwargs):
     rtype = options.pop('rtype', None)
     is_method = options.pop('is_method', True)
     allow_extra = options.pop('allow_extra', False)
-    val_copy = __sort_args(_copy.deepcopy(validator))
```

Comment:
The sorting performed on this line is unnecessary given sorting in lines 428-431 below. The deepcopy is also unnecessary given the other changes.